xls/dslx/run_routines/run_routines_test.cc

```diff
@@ -190,6 +192,161 @@ fn trivial(x: u2) -> bool { id(true) }
   EXPECT_EQ(jit_comparator.jit_cache_.begin()->first, "__test__trivial");
 }
 
+TEST_P(RunRoutinesTest, QuickcheckExhaustiveEnumWithFail) {
+  constexpr const char* kProgram = R"(
+enum MyEnum: u2 {
+  A = 0,
+  B = 1,
+  C = 2,
+}
+
+#[quickcheck(exhaustive)]
+fn qc(x: MyEnum) -> bool {
+    match x {
+        MyEnum::A | MyEnum::B | MyEnum::C => true,
+        _ => fail!("impossible_value", false),
+    }
+}
+)";
+  constexpr const char* kModuleName = "test";
+  constexpr const char* kFilename = "test.x";
+  RunComparator jit_comparator(CompareMode::kJit);
+  ParseAndTestOptions options;
+  options.run_comparator = &jit_comparator;
+  XLS_ASSERT_OK_AND_ASSIGN(
+      TestResultData result,
+      ParseAndTest(kProgram, kModuleName, kFilename, options));
+  EXPECT_THAT(result, IsTestResult(TestResult::kAllPassed, 1, 0, 0));
+
+  ASSERT_EQ(jit_comparator.jit_cache_.size(), 1);
+  EXPECT_EQ(jit_comparator.jit_cache_.begin()->first, "__test__qc");
+}
+
+TEST_P(RunRoutinesTest, EmptyEnum) {
+  constexpr const char* kProgram = R"(
+enum EmptyEnum: u2 {
+}
+
+#[quickcheck(exhaustive)]
+fn qc(x: EmptyEnum) -> bool {
+    true
+}
+)";
+  constexpr const char* kModuleName = "test";
+  constexpr const char* kFilename = "test.x";
+  RunComparator jit_comparator(CompareMode::kJit);
+  ParseAndTestOptions options;
+  options.vfs_factory = [kProgram] {
+    return std::make_unique<UniformContentFilesystem>(kProgram, "test.x");
+  };
+  options.run_comparator = &jit_comparator;
+  XLS_ASSERT_OK_AND_ASSIGN(
+      TestResultData result,
+      ParseAndTest(kProgram, kModuleName, kFilename, options));
+  EXPECT_THAT(result, IsTestResult(TestResult::kSomeFailed, 1, 0, 1));
+  ASSERT_EQ(result.failures().size(), 1);
+  std::string_view failure_message = result.failures()[0];
```

Comment:
This is a lifetime error. 'result.failures()' returns a vector<std::string> so the string_view is pointing at freed memory.

Please make std::string instead

Reply:
This is fixed internally  and should land shortly. No need to update the PR

Reply:
Thanks -- I'll make a followup PR to rename `failures()` since it looks like a span-style getter, lowercase methods aren't supposed to do meaningful work.

Reply:
https://github.com/google/xls/pull/1878